src/states/node.rs

```diff
@@ -2458,3 +2476,336 @@ fn create_first_section_info(public_id: PublicId) -> Result<SectionInfo, Routing
         err
     })
 }
+
+#[cfg(all(test, feature = "mock_parsec"))]
+mod tests {
+    use super::*;
+    use crate::cache::NullCache;
+    use crate::mock_crust::crust::Config;
+    use crate::mock_crust::{self, Network};
+    use crate::outbox::{EventBox, EventBuf};
+    use crate::state_machine::{State, StateMachine};
+    use crate::xor_name::XOR_NAME_LEN;
+    use utils::LogIdent;
+
+    const ACCUMULATE_VOTE_COUNT: usize = 3;
+    const NOT_ACCUMULATE_ALONE_VOTE_COUNT: usize = 2;
+
+    struct NoteUnderTest {
+        pub machine: StateMachine,
+        pub full_id: FullId,
+        pub other_full_ids: Vec<FullId>,
+        pub other_parsec_map: Vec<ParsecMap>,
```

Comment:
I'd maybe merge these two fields into one: `other_nodes: Vec<(FullId, ParsecMap)>` (or better use a struct instead of tuple) to make sure they never get out of sync.

Reply:
I think that would make the use of them less practical/readable, I'm not sure it is a good trade-off for test code.

Reply:
OK, your call.